core-bundle/src/Routing/Enhancer/InputEnhancer.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Enhancer;
+
+use Contao\Config;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\Input;
+use Contao\PageModel;
+use Symfony\Cmf\Component\Routing\Enhancer\RouteEnhancerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+
+class InputEnhancer implements RouteEnhancerInterface
+{
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    /**
+     * @var Input
+     */
+    private $inputAdapter;
+
+    /**
+     * @var Config
+     */
+    private $configAdapter;
+
+    public function __construct(ContaoFrameworkInterface $framework)
+    {
+        $this->framework = $framework;
+
+        $this->inputAdapter = $this->framework->getAdapter(Input::class);
+        $this->configAdapter = $this->framework->getAdapter(Config::class);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function enhance(array $defaults, Request $request)
+    {
+        if (!isset($defaults['pageModel']) || !$defaults['pageModel'] instanceof PageModel) {
+            return $defaults;
+        }
+
+        $this->framework->initialize();
+
+        if (!empty($defaults['_locale']) && $this->configAdapter->get('addLanguageToUrl')) {
+            $this->inputAdapter->setGet('language', $defaults['_locale']);
+        }
+
+        if (empty($defaults['parameters'])) {
+            return $defaults;
+        }
+
+        // TODO should we use trim() instead of substr()?
+        $fragments = explode('/', substr($defaults['parameters'], 1));
+
+        // Add the second fragment as auto_item if the number of fragments is even
+        if ($this->configAdapter->get('useAutoItem') && 0 !== \count($fragments) % 2) {
+            array_unshift($fragments, 'auto_item');
+        }
+
+        for ($i = 0, $c = \count($fragments); $i < $c; $i += 2) {
+            // Skip key value pairs if the key is empty (see #4702)
+            // TODO why does this make sense?
```

Comment:
I guess that is the double slash case `news//something.html`. I will check if the functional tests handle this case.

Reply:
see #263